File: 2277-count-equal-and-divisible-pairs-in-an-array/2277-count-equal-and-divisible-pairs-in-an-array.cs
public class Solution {
    public int CountPairs(int[] nums, int k) {
        int res = 0, n = nums.Length;
        for (int i = 0; i < n; i++) {
            for (int j = i + 1; j < n; j++) {
                if (nums[i] == nums[j] && (i * j) % k == 0) res++;
            }
        }
        return res;
    }
}

File: 2277-count-equal-and-divisible-pairs-in-an-array/2277-count-equal-and-divisible-pairs-in-an-array.cpp
class Solution {
public:
    int countPairs(vector<int>& nums, int k) {
        int ans = 0;
        unordered_map<int,vector<int>>mp;
        for(int i=0;i<nums.size();i++){
            mp[nums[i]].push_back(i);
        }
        for(auto &pair : mp){
            if(pair.second.size() < 2)continue;
            if(pair.second.size() == 2){
                if((pair.second[0] * pair.second[1]) % k == 0)ans++;
            }
            else{
                for(int i=0;i<pair.second.size();i++){
                    for(int j=i+1;j<pair.second.size();j++){
                        if((pair.second[i] * pair.second[j]) % k == 0)ans++;
                    }
                }
            }
        }
        return ans;
    }
};